firefox-ios/Client/Application/BackgroundNotificationSurfaceUtility.swift

```diff
@@ -56,7 +56,13 @@ class BackgroundNotificationSurfaceUtility: BackgroundUtilityProtocol {
     // MARK: Private
     private func setUp() {
         BGTaskScheduler.shared.register(forTaskWithIdentifier: taskIdentifier, using: nil) { task in
-            self.handleAppRefresh(task: task as! BGAppRefreshTask)
+            guard let backgroundRefreshTask = task as? BGAppRefreshTask else {
+                self.logger.log("Failed to cast task to BGAppRefreshTask",
+                                level: .fatal,
+                                category: .lifecycle)
```

Comment:
This PR adds several new `.fatal` logs; I think that's fine, since these would be crashes anyway. But @cyndichin @adudenamedruby please LMK if we have an official policy (or team plan) on how we're updating these force-cast violations.